BotProject/Templates/CSharp/SkillConversationIdFactory.cs

```diff
@@ -0,0 +1,34 @@
+﻿using System.Collections.Concurrent;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Bot.Builder.Skills;
+using Microsoft.Bot.Schema;
+using Newtonsoft.Json;
+
+namespace Microsoft.Bot.Builder.ComposerBot.Json
+{
+    public class SkillConversationIdFactory : SkillConversationIdFactoryBase
+    {
+        private readonly ConcurrentDictionary<string, string> _conversationRefs = new ConcurrentDictionary<string, string>();
```

Comment:
This is in memory, you probably want to use ConversationState or some other persistent state